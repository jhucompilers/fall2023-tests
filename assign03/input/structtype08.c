struct Node {
  int data;
  struct Node *next;
};

int main(void) {
  struct Node n1, n2, n3;

  n1.data = 1;
  n1.next = &n2;

  n2.data = 2;
  n2.next = &n3;

  n3.data = 3;
  n3.next = &n1;

  return 0;
}
