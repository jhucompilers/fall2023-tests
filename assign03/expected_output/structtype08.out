0|struct Node|type|struct Node {}
1|data|variable|int
1|next|variable|pointer to struct Node {}
0|main|function|function () returning int
2|n1|variable|struct Node {int, pointer to struct Node}
2|n2|variable|struct Node {int, pointer to struct Node}
2|n3|variable|struct Node {int, pointer to struct Node}
